LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2EventListener_Soldiers.uc

```diff
@@ -251,11 +251,18 @@ static protected function EventListenerReturn OnOverridePersonnelStatus(Object E
 		{
 			SetStatusTupleData(OverrideTuple, class'UIUtilities_Strategy'.default.m_strOnMissionStatus, "", "", 0, eUIState_Highlight, true);
 		}
+		else if (UnitState.GetRank() < class'XComGameState_LWOutpost'.default.REQUIRED_RANK_FOR_LIAISON_DUTY)
+		{
+			if (GetScreenOrChild('UIPersonnel_Liaison') != none)
+			{
+				SetStatusTupleData(OverrideTuple, default.RankTooLow, "", "", 0, eUIState_Bad, true);
+			}
+		}
 		else if (SquadMgr != none && SquadMgr.UnitIsInAnySquad(UnitState.GetReference(), Squad))
 		{
 			if (SquadMgr.LaunchingMissionSquad.ObjectID != Squad.ObjectID)
 			{
-				if (UnitState.GetStatus() != eStatus_Healing && UnitState.GetStatus() != eStatus_Training)
+				if (UnitState.GetStatus() != eStatus_Healing && UnitState.GetStatus() != eStatus_Training && UnitState.GetMentalState() != eMentalState_Shaken)
```

Comment:
I wonder whether this might as well just check for `eStatus_Active`. Otherwise it also catches soldiers in `eStatus_PsiTesting`/`eStatus_PsiTraining`. And I don't think a soldier can be Shaken and active, though I'm not sure.